stac_fastapi/types/stac_fastapi/types/rfc3339.py

```diff
@@ -45,53 +46,88 @@ def rfc3339_str_to_datetime(s: str) -> datetime:
     return iso8601.parse_date(s)
 
 
-def str_to_interval(interval: Optional[str]) -> Optional[DateTimeType]:
-    """Extract a tuple of datetimes from an interval string.
+def parse_single_date(date_str: str) -> datetime:
+    """
+    Parse a single RFC3339 date string into a datetime object.
 
-    Interval strings are defined by
-    OGC API - Features Part 1 for the datetime query parameter value. These follow the
-    form '1985-04-12T23:20:50.52Z/1986-04-12T23:20:50.52Z', and allow either the start
-    or end (but not both) to be open-ended with '..' or ''.
+    Args:
+        date_str (str): A string representing the date in RFC3339 format.
+
+    Returns:
+        datetime: A datetime object parsed from the date_str.
+
+    Raises:
+        ValueError: If the date_str is empty or contains the placeholder '..'.
+    """
+    if ".." in date_str or not date_str:
+        raise ValueError("Invalid date format.")
```

Comment:
why not raise 400 here too ?

Reply:
the only place this code is accessed is down below via parse_single_date - 
```
try:
        start = parse_single_date(values[0]) if values[0] not in ["..", ""] else None
        end = (
            parse_single_date(values[1])
            if len(values) > 1 and values[1] not in ["..", ""]
            else None
        )
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))